File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-go/Mailstandard/provider_package_catalog.go
package Mailstandard

type MailProviderPackageCatalogEntry struct {
    ProviderKey         string
    PluginId            string
    DriverId            string
    PackageIdentity     string
    ManifestPath        string
    ReadmePath          string
    SourcePath          string
    SourceSymbol        string
    Builtin             bool
    RootPublic          bool
    Status              string
    RuntimeBridgeStatus string
}

type MailProviderPackageCatalog struct{}

var OFFICIAL_mail_PROVIDER_PACKAGES = []MailProviderPackageCatalogEntry{
    {ProviderKey: "smtp", PluginId: "Mail-smtp", DriverId: "sdkwork-mail-driver-smtp", PackageIdentity: "github.com/sdkwork/Mail-sdk-provider-smtp", ManifestPath: "providers/Mail-sdk-provider-smtp/go.mod", ReadmePath: "providers/Mail-sdk-provider-smtp/README.md", SourcePath: "providers/Mail-sdk-provider-smtp/provider_package_contract.go", SourceSymbol: "MailProviderSmtpPackageContract", Builtin: true, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
    {ProviderKey: "imap", PluginId: "Mail-imap", DriverId: "sdkwork-mail-driver-imap", PackageIdentity: "github.com/sdkwork/Mail-sdk-provider-imap", ManifestPath: "providers/Mail-sdk-provider-imap/go.mod", ReadmePath: "providers/Mail-sdk-provider-imap/README.md", SourcePath: "providers/Mail-sdk-provider-imap/provider_package_contract.go", SourceSymbol: "MailProviderImapPackageContract", Builtin: true, RootPublic: false, Status: "future-runtime-bridge-only", RuntimeBridgeStatus: "reserved"},
}

func GetMailProviderPackageByProviderKey(providerKey string) *MailProviderPackageCatalogEntry {
    for index := range OFFICIAL_mail_PROVIDER_PACKAGES {
        if OFFICIAL_mail_PROVIDER_PACKAGES[index].ProviderKey == providerKey {
            return &OFFICIAL_mail_PROVIDER_PACKAGES[index]
        }
    }

    return nil
}

func GetMailProviderPackageByPackageIdentity(packageIdentity string) *MailProviderPackageCatalogEntry {
    for index := range OFFICIAL_mail_PROVIDER_PACKAGES {
        if OFFICIAL_mail_PROVIDER_PACKAGES[index].PackageIdentity == packageIdentity {
            return &OFFICIAL_mail_PROVIDER_PACKAGES[index]
        }
    }

    return nil
}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-python/sdkwork_mail_sdk/provider_package_catalog.py
from dataclasses import dataclass


@dataclass(frozen=True)
class MailProviderPackageCatalogEntry:
    providerKey: str
    pluginId: str
    driverId: str
    packageIdentity: str
    manifestPath: str
    readmePath: str
    sourcePath: str
    sourceSymbol: str
    builtin: bool
    rootPublic: bool
    status: str
    runtimeBridgeStatus: str


class MailProviderPackageCatalog:
    entries = [
        MailProviderPackageCatalogEntry("smtp", "Mail-smtp", "sdkwork-mail-driver-smtp", "sdkwork-mail-sdk-provider-smtp", "providers/sdkwork_mail_sdk_provider_smtp/pyproject.toml", "providers/sdkwork_mail_sdk_provider_smtp/README.md", "providers/sdkwork_mail_sdk_provider_smtp/sdkwork_mail_sdk_provider_smtp/__init__.py", "MailProviderSmtpPackageContract", True, False, "future-runtime-bridge-only", "reserved"),
        MailProviderPackageCatalogEntry("imap", "Mail-imap", "sdkwork-mail-driver-imap", "sdkwork-mail-sdk-provider-imap", "providers/sdkwork_mail_sdk_provider_imap/pyproject.toml", "providers/sdkwork_mail_sdk_provider_imap/README.md", "providers/sdkwork_mail_sdk_provider_imap/sdkwork_mail_sdk_provider_imap/__init__.py", "MailProviderImapPackageContract", True, False, "future-runtime-bridge-only", "reserved"),
    ]


def get_mail_provider_package_by_provider_key(provider_key: str) -> Optional[MailProviderPackageCatalogEntry]:
    for entry in MailProviderPackageCatalog.entries:
        if entry.providerKey == provider_key:
            return entry

    return None


def get_mail_provider_package_by_package_identity(package_identity: str) -> Optional[MailProviderPackageCatalogEntry]:
    for entry in MailProviderPackageCatalog.entries:
        if entry.packageIdentity == package_identity:
            return entry

    return None
